trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/FunctionHandler.java

```diff
@@ -716,6 +688,57 @@ private Result handleFunctionCallGivenNameAndArguments(final IDispatcher main, f
 			translatedParams.add(in.getLrValue().getValue());
 			functionCallExpressionResultBuilder.addAllExceptLrValue(in);
 		}
+		if (calleeProcCType != null && calleeProcCType.hasVarArgs()) {
+			final boolean hasUsedVarargs = calleeProcCType.getVarArgsUsage() == VarArgsUsage.USED;
+			// For varargs we need a special handling:
+			// - If the varargs are not used, we simply dispatch the arguments, without passing them to the function.
+			// - If they are used, we create a pointer for all the remaining arguments and pass them to the function.
+			final AuxVarInfo auxvarinfo = mAuxVarInfoBuilder.constructAuxVarInfo(loc,
+					mTypeHandler.constructPointerType(loc), SFO.AUXVAR.VARARGS_POINTER);
+			if (hasUsedVarargs) {
+				functionCallExpressionResultBuilder.addAuxVar(auxvarinfo);
+				functionCallExpressionResultBuilder.addDeclaration(auxvarinfo.getVarDec());
+			}
+			final CPrimitive pointerType = mExpressionTranslation.getCTypeOfPointerComponents();
+			BigInteger currentOffset = BigInteger.ZERO;
+			final List<Statement> writes = new ArrayList<>();
+			final Expression originalBase = MemoryHandler.getPointerBaseAddress(auxvarinfo.getExp(), loc);
+			final Expression originalOffset = MemoryHandler.getPointerOffset(auxvarinfo.getExp(), loc);
+			for (final ExpressionResult va : varargs) {
+				final ExpressionResult param;
+				int size;
+				if (va.getCType().getUnderlyingType() instanceof CPrimitive) {
+					// All smaller types (char, short) are promoted to int
+					param = mExprResultTransformer.doIntegerPromotion(loc, va);

```

Comment:
You don't actually check if the type is smaller. How does this behave on e.g. a `long long`?